cmd/cored/main.go

```diff
@@ -140,11 +136,11 @@ func main() {
 	if err != nil {
 		chainlog.Fatalkv(ctx, chainlog.KeyError, err)
 	}
-	listener, cert, err := maybeUseTLS(listener)
+	listener, tlsConfig, err := maybeUseTLS(listener)
```

Comment:
Actually no: the Server's TLSConfig is used by ListenAndServeTLS, which we don't call. Instead, we make our own *tls.Listener and provide that to Server.Serve. It is confusing. :/